components/Poll/Poll.jsx

```diff
@@ -0,0 +1,342 @@
+        if (selectedIndex === optIndex) {
+          this.serverVote(opts[optIndex], true);
+        } else if (selectedIndex >= 0) {
+          pollVote(this.props.postId, { retract: true, optionId: opts[selectedIndex]._id })
+          .then((poll) => {
+            this.serverVote(opts[optIndex], false);
+          })
+          .catch((error) => alert("Error updating vote. Sorry about that!"));
+        } else {
+          this.serverVote(opts[optIndex], false);
+        }
+      } else {
+        for (var i = 0; i < opts.length; i++) {
+          if (i === optIndex) {
+            this.updateOptionVoters(opts[optIndex]);
+          } else {
+            let userIndex = this.userVoteIndex(opts[i]);
+            if (userIndex >= 0) {
+              opts[i].usersVoted.splice(userIndex, 1);
+            }
+          }
+        }
+      }
+    }
+    this.props.savePoll && this.props.savePoll({
+      title: this.state.title,
+      multiSelect: this.state.multiSelect,
+      open: this.state.open,
+      options: opts,
+    });
+  }
+
+  confirmDelete = async (item) => {
+    if (!this.props.isAuthor || !this.props.savePoll) {
+      return;
+    }
+    Alert.alert(
+      'Hold Up!',
+      `Are you sure you want to remove this option?\nOption: ${item.text}`,
+      [
+        { text: 'Cancel', style: 'cancel' },
+        { text: 'Delete', onPress: this.deleteOption.bind(this, item) },
+      ],
+    );
+  }
+
+  deleteOption = (item) => {
+    var opts = this.state.options;
+    var optIndex = opts.findIndex(opt => opt.key === item.key);
+    if (optIndex === -1) {
+      return;
+    }
+    opts.splice(optIndex, 1);
+    this.props.savePoll({
+      title: this.state.title,
+      multiSelect: this.state.multiSelect,
+      open: this.state.open,
+      options: opts,
+    });
+  }
+
+  addOption = async () => {
+    if (this.state.newOption.length == 0) {
+      return;
+    }
+    let optionText = this.state.newOption;
+    this.setState({ newOption: '' });
+    if (!this.props.savePoll) {
+      pollOption(this.props.postId, { option: optionText })
+      .then((poll) => {
+        this.setState({ options: poll.options });
+        this.props.updatePoll && this.props.updatePoll({
+          title: this.state.title,
+          multiSelect: this.state.multiSelect,
+          open: this.state.open,
+          options: poll.options,
+        });
+      })
+      .catch((error) => alert("Error adding poll option. Sorry about that!"));
+    } else {
+      var opts = this.state.options;
+      opts.push({
+        key: `${this.state.options.length}`,
+        text: optionText,
+        usersVoted: [],
+      });
+      this.props.savePoll({
+        title: this.state.title,
+        multiSelect: this.state.multiSelect,
+        open: this.state.open,
+        options: opts,
+      });
+    }
+  }
+
+  buildListItems = () => {
+    let items = this.state.options
+            .sort((a, b) => b.usersVoted.length - a.usersVoted.length)
```

Comment:
When an option is pressed twice in rapid succession, it produces an error:

TypeError: TypeError: undefined is not an object (evaluating '_this.state.options.sort') 

I can provide you more detail if you aren't able to reproduce the issue yourself.

Reply:
Should be fixed in most recent update